### PR TITLE
fix specaug performance issue

### PR DESCRIPTION
**Pull request by bobchennan**
fixed specaug performance issue caused by commit 68c0351. 
However I still need to fix shape problem and check performance.

Current performance for 0.4.0 with specaug:

System(#iters) | CER(dev) | WER(dev) | CER(eval) | WER(eval)
-- | -- | -- | -- | --
rnn(22) | 3.4 | 7.4 | 2.4 | 5.0
transformer(45) | 3.8 | 8.4 | 2.5 | 5.1

One way to improve it is to set `patience=0` and train with more iterations.
Currently I resumed training of transformer.

Transformer training curve before early-stopping:
image

### CODE REVIEW

File: espnet/asr/pytorch_backend/asr.py [hunk 1]
```diff
@@ -560,7 +565,6 @@ def recog(args):
                         if hyps is not None:
                             text = ''.join([train_args.char_list[int(x)]
                                             for x in hyps[0]['yseq'][1:-1] if int(x) != -1])
-                            text = text.replace('\u2581', ' ').strip()  # for SentencePiece
```
Comment (sw005320) [inline espnet/asr/pytorch_backend/asr.py]: This was introduced in #809. Please recover it.

**Comment by ShigekiKarita** [comment]
Can you confirm results w/o spectaug in your env?

**Comment by bobchennan** [comment]
some updates:

System(#iters) | WER dev93 | WER eval92
-- | -- | --
Transformer | 8.2 | 4.8
Transformer+specaug | 7.3 | 4.6
RNN | 8.6 | 5.1
RNN+specaug | |

File: egs/wsj/asr1/conf/tuning/train_rnn.yaml [hunk 2]
```diff
@@ -36,3 +36,6 @@ patience: 3
 
 # scheduled sampling option
 sampling-probability: 0.0
+
+# SpecAug
+use-specaug: True
```
Comment (ShigekiKarita) [inline egs/wsj/asr1/conf/tuning/train_rnn.yaml]: Can you update RESULTS.md if you wanna make this default

Comment (bobchennan) [reply egs/wsj/asr1/conf/tuning/train_rnn.yaml]: Sure I will update code and RESULTS.md later.

**Comment by bobchennan** [comment]
I am training last exp which disables label smoothing after warmup.

Unless it gives different results, my conclusion is that time warping is really helpful for RNN model since it helps to learn certain gates by disturbing speed rate; while this is not helpful for transformer since it always consider all frames simultaneously. For transformer time and frequency masking improves a little bit. Conclusion might be different for larger corpus since maybe we overfitted too much on dev.

**Comment by bobchennan** [comment]
System(#iters) | WER dev93 | WER eval92
-- | -- | --
Transformer | 8.2 | 4.8
Transformer+FM+TM | 7.3 | 4.6
Transformer+FM+TM+TW | 7.0 | 5.0
Transformer+FM+TM+TW w.o. label smoothing | 7.3 | 4.7
RNN | 8.6 | 5.1
RNN+FM+TM | 7.8 | 5.2
RNN+FM+TM+TW | 7.3 | 4.6

**Comment by ShigekiKarita** [comment]
@bobchennan as you mentioned before, #745 implements specaug as preprocessing. And it has already utilized your `sparse_image_warp` like this
```yaml
process:
  # these three processes are a.k.a. SpecAugument
  - type: "time_warp"
    max_time_warp: 5
    inplace: true
    mode: "sparse_image_warp"  # "PIL" is faster
  - type: "freq_mask"
    F: 30
    n_mask: 2
    inplace: true
    replace_with_zero: false
  - type: "time_mask"
    T: 40
    n_mask: 2
    inplace: true
    replace_with_zero: false
```
see also https://github.com/espnet/espnet/blob/d988a9298479c1550743fbd50334c735e0ab9c65/espnet/transform/spec_augment.py#L41-L44

Therefore I do not think `--use-specaug` option is needed because it is not configurable like `--preprocess-conf`. If you agree, I would like to request these changes 
- remove `use_specaug` related codes in `asr_train.py` and `asr.py`
- add new config `egs/wsj/asr1/conf/specaug_sparse.yaml` with your new setting?

**Comment by bobchennan** [comment]
Sure. #745 is too long so I didn't notice that.

**Comment by bobchennan** [comment]
@ShigekiKarita 
I changed it. It may still be different with results I reported since current implementation of time masking and freq masking is actually different with what I used. Pls check espnet/espnet/transform/spec_augment.py and espnet/utils/spec_augment.py for details.

File: espnet/asr/pytorch_backend/asr.py (hunk 1 above)
Comment (sw005320) [reply espnet/asr/pytorch_backend/asr.py]: @bobchennan, can you fix this?
